mypyc/lib-rt/CPy.h

```diff
@@ -672,6 +672,8 @@ static bool CPyList_SetItem(PyObject *list, CPyTagged index, PyObject *value) {
                 return false;
             }
         }
+        // PyList_SET_ITEM doesn't decref the old element, so we do
+        Py_XDECREF(PyList_GET_ITEM(list, n));
```

Comment:
Why do we need to use `Py_XDECREF` here? Can the item be `NULL`?

Reply:
I did that because the code in PyList_SetItem did, but I suspect that was because of an overabundance of caution. It can be necessary if the list isn't fully initialized yet, but I don't think we need to worry about that.